x/incentives/keeper/gauge_test.go

```diff
@@ -775,14 +673,168 @@ func (s *KeeperTestSuite) TestInitGaugeInfo() {
+			poolIDs:          []uint64{poolInfo.ConcentratedPoolID, poolInfo.BalancerPoolID, poolInfo.StableSwapPoolID},
+
+			expectedPerpeutalGroupGauge: false, // explicit for clarity
+			expectedGaugeInfo: addGaugeRecords(defaultEmptyGaugeInfo, []types.InternalGaugeRecord{
+				concentratedGaugeRecord,
+				balancerGaugeRecord,
+				stableSwapGaugeRecord,
+			}),
+		},
+
+		// error cases
+
+		{
+			name:             "error: fails to initialize group gauge due to cosmwasm pool that does not support incentives",
+			coins:            defaultCoins,
+			numEpochPaidOver: incentiveskeeper.PerpetualNumEpochsPaidOver,
+			poolIDs:          []uint64{poolInfo.BalancerPoolID, poolInfo.CosmWasmPoolID},
+
+			expectErr: poolincentivetypes.UnsupportedPoolTypeError{PoolID: poolInfo.CosmWasmPoolID, PoolType: poolmanagertypes.CosmWasm},
+		},
+
+		{
+			name:                "error: owner does not have enough funds",
+			coins:               defaultCoins,
+			creatorAddressIndex: fundedAddressIndex + 1,
+			numEpochPaidOver:    incentiveskeeper.PerpetualNumEpochsPaidOver,
+			poolIDs:             []uint64{poolInfo.BalancerPoolID, poolInfo.ConcentratedPoolID},
+			expectErr:           fmt.Errorf("0uosmo is smaller than %s: insufficient funds", defaultCoins),
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+
+			// Alwats fund the first account
+			s.FundAcc(s.TestAccs[fundedAddressIndex], tc.coins)
+
+			groupGaugeId, err := s.App.IncentivesKeeper.CreateGroup(s.Ctx, tc.coins, tc.numEpochPaidOver, s.TestAccs[tc.creatorAddressIndex], tc.poolIDs)
+			if tc.expectErr != nil {
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, tc.expectErr.Error())
+			} else {
+				s.Require().NoError(err)
+
+				s.Require().Equal(expectedGroupGaugeId, groupGaugeId)
+
+				// Validate group's Gauge
+				groupGauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, groupGaugeId)
+				s.Require().NoError(err)
+				s.Require().Equal(tc.coins, groupGauge.Coins)
+				s.Require().Equal(tc.numEpochPaidOver, groupGauge.NumEpochsPaidOver)
+				s.Require().Equal(tc.expectedPerpeutalGroupGauge, groupGauge.IsPerpetual)
+				s.Require().Equal(lockuptypes.ByGroup, groupGauge.DistributeTo.LockQueryType)
+
+				// Validate Group
+				group, err := s.App.IncentivesKeeper.GetGroupByGaugeID(s.Ctx, groupGaugeId)
+				s.Require().NoError(err)
+
+				s.Require().Equal(expectedGroupGaugeId, group.GroupGaugeId)
+				s.Require().Equal(types.ByVolume, group.SplittingPolicy)
+
+				// Validate InternalGaugeInfo
+				actualGaugeInfo := group.InternalGaugeInfo
+				s.validateGaugeInfo(tc.expectedGaugeInfo, actualGaugeInfo)
+
+				// Bump up expected gauge ID since we are reusing the same test state
+				expectedGroupGaugeId++
 			}
 		})
 	}
 }
+
+// validates that the expected gauge info equals the actual gauge info
+func (s *KeeperTestSuite) validateGaugeInfo(expected types.InternalGaugeInfo, actual types.InternalGaugeInfo) {
```

Comment:
Note: created helper at the bottom per discussions in: https://github.com/osmosis-labs/osmosis/issues/6520